docs/client_libraries.md

```diff
@@ -0,0 +1,286 @@
+container ensure you have access to NGC.  For step-by-step
+instructions, see the [NGC Getting Started
+Guide](http://docs.nvidia.com/ngc/ngc-getting-started-guide/index.html).
+
+Use docker pull to get the client libraries and examples container
+from NGC.
+
+```bash
+$ docker pull nvcr.io/nvidia/tritonserver:<xx.yy>-py3-clientsdk
+```
+
+Where <xx.yy> is the version that you want to pull. Within the
+container the client libraries are in /workspace/install/lib, the
+corresponding headers in /workspace/install/include, and the Python
+wheel files in /workspace/install/python. The image will also contain
+the built client examples that you can learn more about in
+[Examples](client_examples.md).
+
+### Build Using Docker
+
+To build the client libraries using Docker, first change directory to
+the root of the repo and checkout the release version of the branch
+that you want to build (or the master branch if you want to build the
+under-development version). The branch you use for the client build
+should match the version of Triton you are using.
+
+```bash
+$ git checkout r20.09
+```
+
+Then, issue the following command to build the C++ client library and
+the Python wheel files for the Python client library.
+
+```bash
+$ docker build -t tritonserver_client -f Dockerfile.client .
+```
+
+You can optionally add *--build-arg "BASE_IMAGE=<base_image>"* to set
+the base image that you want the client library built against. This
+base image must be an Ubuntu CUDA image to be able to build CUDA
+shared memory support. If CUDA shared memory support is not required,
+you can use Ubuntu 18.04 as the base image.
+
+After the build completes the tritonserver_client docker image will
+contain the built client libraries in /workspace/install/lib, the
+corresponding headers in /workspace/install/include, and the Python
+wheel files in /workspace/install/python. The image will also contain
+the built client examples that you can learn more about in
+[Examples](client_examples.md).
+
+### Build Using CMake
+
+The client library build is performed using CMake. The build
+dependencies and requirements are shown in Dockerfile.client. To build
+without Docker you must first install those dependencies. This section
+describes the client build for Ubuntu 18.04 and Windows 10
+systems.
+
+To build the libraries using CMake, first change directory to the root
+of the repo and checkout the release version of the branch that you
+want to build (or the master branch if you want to build the
+under-development version).
+
+```bash
+$ git checkout r20.09
+```
+
+#### Ubuntu 18.04
+
+For Ubuntu, the dependencies and how to install them can be found in
+Dockerfile.client. Also note that the dependency name may be different
+depending on the version of the system.
+
+To build on Ubuntu, run the following to configure and build:
+
+```bash
+$ mkdir builddir && cd builddir
+$ cmake -DCMAKE_BUILD_TYPE=Release ../build
```

Comment:
I think we should mention dependency on the common and core repos for the client library build/s.